src/RepoView.tsx

```diff
@@ -91,53 +91,54 @@ const RepoView = (props: RepoViewProps) => {
       setSelectedIndex(index);
       return false;
     });
-    invoke("get_commit_history").then((result) => {
+    invoke('get_commit_history').then((result) => {
       setBranchCommits(result as Commit[]);
     });
-  }, []);
+  }, [repo.branchNames]);
```

Comment:
resolved from a `rules-of-hooks` error. gotta watch those dependencies! (in this case, fixes a bug where updates to `props.repo.branchNames` array would not properly trigger the `useEffect`, meaning the selected indicies would not be updated properly 😝